src/scf.f90

```diff
@@ -528,6 +528,10 @@ subroutine electdiis(jscf)
 
          call CopyDMat(quick_qm_struct%dense,quick_scratch%hold,nbasis) ! Save DENSE to HOLD
 
+#if defined(CUDA) || defined(CUDA_MPIV)
+         call gpu_get_dmx(quick_qm_struct%co, quick_qm_struct%dense)
+#else
+
          ! PIJ=SIGMA[i=1,nelec/2]2CJK*CIK
          do I=1,nbasis
```

Comment:
This old code to compute the density matrix is very inefficient. No cache-blocking is employed. It is limited by memory bandwidth.